src/test/java/org/broadinstitute/hellbender/cmdline/GATKPlugin/GATKAnnotationPluginDescriptorUnitTest.java

```diff
@@ -0,0 +1,460 @@
+                Collections.emptySet());
+        String[] args = {"-G", StandardAnnotation.class.getSimpleName()};
+        clp.parseArguments(nullMessageStream, args);
+        List<Annotation> annots = instantiateAnnotations(clp);
+        final VariantAnnotatorEngine vae = new VariantAnnotatorEngine(annots, null, Collections.emptyList(), false);
+
+        Assert.assertFalse(vae.getGenotypeAnnotations().isEmpty());
+        Assert.assertFalse(vae.getInfoAnnotations().isEmpty());
+
+        //check that Coverage is in and ClippingRankSum is out
+        Assert.assertTrue(vae.getInfoAnnotations().stream().anyMatch(a -> a.getClass().getSimpleName().equals(Coverage.class.getSimpleName())));
+        Assert.assertTrue(vae.getInfoAnnotations().stream().anyMatch(a -> a.getClass().getSimpleName().equals(ClippingRankSumTest.class.getSimpleName())));
+    }
+
+    @DataProvider
+    public Object[][] groupHierarchyArguments(){
+        return new Object[][]{
+                // we must add --DISABLE_TOOL_DEFAULT_ANNOTATIONS because we pass our annotation objects in to have their groups dynamically discovered
+                {new String[]{"--"+ StandardArgumentDefinitions.DISABLE_TOOL_DEFAULT_ANNOTATIONS}, false, false},
+                {new String[]{"--"+ StandardArgumentDefinitions.DISABLE_TOOL_DEFAULT_ANNOTATIONS, "-G","ParentAnnotationGroup"}, true, true},
+                {new String[]{"--"+ StandardArgumentDefinitions.DISABLE_TOOL_DEFAULT_ANNOTATIONS, "-G","ChildAnnotationGroup"}, true, false},
+                {new String[]{"--"+ StandardArgumentDefinitions.DISABLE_TOOL_DEFAULT_ANNOTATIONS, "-G","ParentAnnotationGroup","-G","ChildAnnotationGroup"}, true, true},
+                {new String[]{}, true, true},};
+    }
+
+    // Though current annotation groups aren't implemented in a hierarchical manner, this test enforces that they would behave
+    // in an expected manner as well as testing that annotation groups can be dynamically discovered
+    @Test (dataProvider = "groupHierarchyArguments")
+    public void testGroupHierarchyBehavior(String[] args, boolean includeChild, boolean includeParent){
+        testChildAnnotation childAnnotation = new testChildAnnotation();
+        testParentAnnotation parentAnnotation = new testParentAnnotation();
+        CommandLineParser clp = new CommandLineArgumentParser(
+                new Object(),
+                Collections.singletonList(new GATKAnnotationPluginDescriptor(Arrays.asList(parentAnnotation, childAnnotation), null)),
+                Collections.emptySet());
+        clp.parseArguments(nullMessageStream, args);
+        List<Annotation> annots = instantiateAnnotations(clp);
+        Assert.assertEquals(annots.contains(childAnnotation), includeChild);
+        Assert.assertEquals(annots.contains(parentAnnotation), includeParent);
+    }
+
+    private interface ParentAnnotationGroup extends Annotation { }
+    private interface ChildAnnotationGroup extends ParentAnnotationGroup { }
```

Comment:
I know we don't have a good solution to this problem, but these classes aren't discoverable by the descriptor, so the tests that rely on them aren't really exercising the underlying code (i.e., because they're not discoverable, they don't expose the issue with redundant copies of an annotation being added to the group list).